UBCFormulaElectric/Consolidated-Firmware
# Implement JSON CAN faults and warnings system

### Summary

Add CAN faults and warnings which centralizes setting alerts over CAN. Faults will cause the BMS to open contactors and enter fault state, and warnings will be displayed to the driver via the DIM's LEDs (but board's won't do any response).

### Changelist 
- Change all alerts and warnings to use the new CAN alerts system
- Update jsoncan submodule 
- Update tests where necessary

### Testing Done

(all tests run on DIM since the CAN alerts system is identical between boards, and the DIM is the only one to check if any other boards set any warnings/faults)

Test 1: Warning TX (PASSED)
1. DIM sets a few warnings and confirm that the corresponding warning signal is transmitted over CAN

Test 2: Fault TX (PASSED)
1. DIM sets a few faults and confirm that the corresponding fault signal is transmitted over CAN

Test 3: Warning/Fault RX (PASSED)
1. Send the DIM a few warnings from different boards and confirm its received correctly (correct LED is enabled)
2. Send the DIM a few faults from different boards and confirm its received correctly (correct LED is enabled)

### Resolved Issues


### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
